File: smartreceptionsdkdemojava/src/main/java/com/ocm/smartreceptionsdkdemojava/MacHelper.java
package com.ocm.smartreceptionsdkdemojava;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.telephony.TelephonyManager;

import java.io.IOException;
import java.io.InputStreamReader;
import java.io.LineNumberReader;

public class MacHelper {
    private static int getAPNType(Context context) {
        int netType = 0;
        ConnectivityManager connMgr = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
        if (networkInfo == null) {
            return netType;
        }
        int nType = networkInfo.getType();
        if (nType == ConnectivityManager.TYPE_WIFI) {
            netType = 1;// wifi
        } else if (nType == ConnectivityManager.TYPE_MOBILE) {
            int nSubType = networkInfo.getSubtype();
            TelephonyManager mTelephony = (TelephonyManager) context
                    .getSystemService(Context.TELEPHONY_SERVICE);
            if (nSubType == TelephonyManager.NETWORK_TYPE_UMTS
                    && !mTelephony.isNetworkRoaming()) {
                netType = 2;// 3G
            } else {
                netType = 3;// 2G
            }
        }else{
            netType = 4;
        }
        return netType;
    }

    public static String getLocalMac(Context context) {
        String mac=null;
        String str = "";
        String str1 = "cat /sys/class/net/wlan0/address";
        try
        {
            Process pp = null;
            try {
                pp = Runtime.getRuntime().exec(str1);
            } catch (IOException e) {
                e.printStackTrace();
            }
            InputStreamReader ir = new InputStreamReader(pp.getInputStream());
            LineNumberReader input = new LineNumberReader(ir);
            for (; null != str;)
            {
                str = input.readLine();
                if (str != null)
                {
                    mac = str.trim();// 去空格
                    break;
                }
            }
        } catch (IOException ex) {
            // 赋予默认值
            ex.printStackTrace();
        }
        return mac;
    }
}

File: app/src/main/java/com/ocm/smartreceptionsdkdemo/MacHelper.kt
package com.ocm.smartreceptionsdkdemo

import android.content.Context
import android.net.ConnectivityManager
import android.telephony.TelephonyManager
import java.io.IOException
import java.io.InputStreamReader
import java.io.LineNumberReader

object MacHelper {
    private fun getAPNType(context: Context): Int {
        var netType = 0
        val connMgr = context
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = connMgr.activeNetworkInfo ?: return netType
        val nType = networkInfo.type
        if (nType == ConnectivityManager.TYPE_WIFI) {
            netType = 1// wifi
        } else if (nType == ConnectivityManager.TYPE_MOBILE) {
            val nSubType = networkInfo.subtype
            val mTelephony = context
                    .getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
            if (nSubType == TelephonyManager.NETWORK_TYPE_UMTS && !mTelephony.isNetworkRoaming) {
                netType = 2// 3G
            } else {
                netType = 3// 2G
            }
        } else {
            netType = 4
        }
        return netType
    }

    fun getLocalMac(context: Context): String? {
        var mac: String? = null
        var str: String? = ""
        val str1 = "cat /sys/class/net/wlan0/address"
        try {
            var pp: Process? = null
            try {
                var cmdStr = str1
                pp = Runtime.getRuntime().exec(cmdStr)
            } catch (e: IOException) {
                e.printStackTrace()
            }

            val ir = InputStreamReader(pp!!.inputStream)
            val input = LineNumberReader(ir)
            while (null != str) {
                str = input.readLine()
                if (str != null) {
                    mac = str.trim { it <= ' ' }// 去空格
                    break
                }
            }
        } catch (ex: IOException) {
            // 赋予默认值
            ex.printStackTrace()
        }

        return mac
    }
}
